### DISCUSSION TITLE
Förändringar i lagersaldo och identifiering av produkt

### DISCUSSION THREAD

**Discussion by ekiminatorn** [CNA-22/products-service#4] [General]
@CNA-22/product-service 
@CNA-22/inventory-service 
@CNA-22/shipping-service 
@CNA-22/order-service 
@CNA-22/frontend-store 
@CNA-22/frontend-admin
@CNA-22/cart-service 

****FYI** för alla som har något med produkter eller lagersaldo att göra:**

I fortsättningen frågas lagersaldo rakt från lagret (inventory API), inte via products API. Produkt och saldo kommer att förknippas ihop med hjälp av product ID (pid) som genereras av products API vid skapande av ny produkt. 

D.v.s. Om man vill veta lagersaldo för en viss produkt, ska man använda pid för att skicka förfrågan till inventory API.

Frågor eller bekymmer angående detta kan ställas här.

**Comment by FrediWa** [comment]
Är lagersaldot något frontenden behöver bry om sig? Eller är saldot något backenden kollar före den ens kommer till frontenden?

I vilket skede kollas saldot?
1) När produkterna listas
2) När produkten tillsätts till kärran
3) När man gör beställningen

**Reply by ekiminatorn** [reply]
Ursäkta mig att det har tagit såhär länge att svara.

Products API kommer inte att kolla lagersaldo, utan meningen är att de kollas skilt av er. 

Det vore nice för kunden om produkten är tillfälligt slut så skulle de visas vid listningen av produkter. Detta kräver ju då att man borda fråga skilt för varje produkt vad lagersaldon är. Problemet med de kommer säkert att ta en stund före man har fått svar på alla lagersaldo förfrågan. Det var tal att backenden skulle cache lagersaldo, men om jag kommer rätt ihåg, så skippade vi de för att hålla projektet simpelt.

Frontend skulle kunna göra en simpel lagersaldo check vid tillsättande till kärran och kasta en error message om lagret är tomt.
